s2spy/time.py

```diff
@@ -90,430 +103,308 @@ def __init__(
 
-    def _repr_html_(self):
-        """For jupyter notebook to load html compatiable version of __repr__."""
-        if self._intervals is not None:
-            # pylint: disable=protected-access
-            return self._label_targets(self._intervals)._repr_html_()
+        Args:
+            interval (pd.Interval): Pandas interval.
 
-        props = ", ".join(
-            [f"{k}={v}" for k, v in self.__dict__.items() if not k.startswith("_")]
-        )
-        return f"AdventCalendar({props})"
-
-    @property
-    def flat(self):
-        if self._intervals is not None:
-            return self._intervals.stack()
-        raise ValueError("The calendar is not initialized with intervals yet."
-                         "use `map_years` or `map_to_data` methods to set it up.")
-
-    def discard(self, max_lag):  # or "set_max_lag"
-        """Only keep indices up to the given max lag."""
-        # or think of a nicer way to discard unneeded info
-        raise NotImplementedError
-
-    def _mark_target_period(
-        self, input_data: Union[pd.DataFrame, xr.Dataset]
-    ) -> Union[pd.DataFrame, xr.Dataset]:
-        """Mark interval periods that fall within the given number of target periods.
-
-        Pass a pandas Series/DataFrame with an 'i_interval' column, or an xarray
-        DataArray/Dataset with an 'i_interval' coordinate axis. It will return an
-        object with an added column in the Series/DataFrame or an
-        added coordinate axis in the DataSet called 'target'. This is a boolean
-        indicating whether the index time interval is a target period or not. This is
-        determined by the instance variable 'n_targets'.
+        Returns:
+            str: String in the form of '(2020-50, 2020-51]'
+        """
+        left = interval.left.strftime("%Y-W%W")
+        right = interval.right.strftime("%Y-W%W")
+        return f"({left}, {right}]"
 
-        Args:
-            input_data: Input data for resampling. For a Pandas object, one of its
-            columns must be called 'i_interval'. An xarray object requires a coordinate
-            axis named 'i_interval' containing an interval counter for every period.
+    def show(self) -> pd.DataFrame:
+        """Displays the intervals the Calendar will generate for the current setup.
 
         Returns:
-            Input data with boolean marked target periods, similar data format as
-                given inputs.
+            pd.Dataframe: Dataframe containing the calendar intervals, with the target
+                periods labelled.
         """
-        if isinstance(input_data, PandasData):
-            input_data['target'] = np.zeros(input_data.index.size, dtype=bool)
-            input_data['target'] = input_data['target'].where(
-               input_data['i_interval'] >= self._n_targets, other=True)
+        return self._label_targets(self.get_intervals()).applymap(
+            self._interval_as_weeknr
+        )
+
+
+def resample(
```

Comment:
Wouldn't it make more sense for the resample function to be placed with the rest of the implementation inside _resample.py?